syntax = "proto3";

import "google/protobuf/timestamp.proto";
import "proto/data/review.proto";

package b4youeat;

service ReviewService {
  rpc CreateReview(CreateReviewRequest) returns (CreateReviewResponse);
  rpc GetReview(GetReviewRequest) returns (GetReviewResponse);
}


message GetReviewRequest {
  int32 id = 1;
}

message GetReviewResponse{
  b4youeat.review.Review review = 1;
}

message CreateReviewRequest {
  int32 restaurant_id = 1;
  int32 user_id = 2;
  google.protobuf.Timestamp visit_date = 3;
  string content = 4;
  int32 rating = 5;
}

message CreateReviewResponse {
  b4youeat.review.Review review = 1;
}

